_studio/shared/src/libmfx_allocator_vaapi.cpp

```diff
@@ -203,18 +241,17 @@ static mfxStatus ReallocImpl(mfxDefaultAllocatorVAAPI::mfxWideHWFrameAllocator*
                             attrCnt,
                             &attrib[0], attrCnt);
 
-    *vaapi_mid->m_surface = surfaces[0];
-    vaapi_mid->m_fourcc = fourcc;
-
     if (VA_STATUS_SUCCESS != va_res)
     {
         return MFX_ERR_DEVICE_FAILED;
```

Comment:
Here (when return) u left undefined `m_surface` because it was destroyed at #229. I believe logic should be - destroy & invalidate `m_surface` (set to `VA_INVALID_ID`?), then create new surface and if all is OK assign it to `m_surface`.